bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/presence/NAPresenceCameraHandler.java

```diff
@@ -0,0 +1,175 @@
+    }
+
+    private State getFloodlightState() {
+        if (module != null) {
+            final boolean isOn = NAWelcomeCamera.LightModeStatusEnum.ON.equals(module.getLightModeStatus());
+            return toOnOffType(isOn);
+        }
+        return UnDefType.UNDEF;
+    }
+
+    private State getFloodlightAutoModeState() {
+        if (module != null) {
+            return toOnOffType(NAWelcomeCamera.LightModeStatusEnum.AUTO.equals(module.getLightModeStatus()));
+        }
+        return UnDefType.UNDEF;
+    }
+
+    private void switchFloodlight(boolean isOn) {
+        if (isOn) {
+            changeFloodlightMode(NAWelcomeCamera.LightModeStatusEnum.ON);
+        } else {
+            switchFloodlightAutoMode(OnOffType.ON.equals(floodlightAutoModeState));
+        }
+    }
+
+    private void switchFloodlightAutoMode(boolean isAutoMode) {
+        floodlightAutoModeState = toOnOffType(isAutoMode);
+        if (isAutoMode) {
+            changeFloodlightMode(NAWelcomeCamera.LightModeStatusEnum.AUTO);
+        } else {
+            changeFloodlightMode(NAWelcomeCamera.LightModeStatusEnum.OFF);
+        }
+    }
+
+    private void changeFloodlightMode(NAWelcomeCamera.LightModeStatusEnum mode) {
+        Optional<String> localCameraURL = getLocalCameraURL();
+        if (localCameraURL.isPresent()) {
+            String url = localCameraURL.get()
+                    + FLOODLIGHT_SET_URL_PATH
+                    + "?config=%7B%22mode%22:%22"
+                    + mode.toString()
+                    + "%22%7D";
+            executeGETRequest(url);
+        }
+    }
+
+    private Optional<String> getLocalCameraURL() {
+        String vpnURL = getVpnUrl();
+        if (vpnURL != null) {
+            //The local address is (re-)requested when it wasn't already determined or when the vpn address was changed.
+            if (!cameraAddress.isPresent() || cameraAddress.get().isVpnURLChanged(vpnURL)) {
+                Optional<JSONObject> json = executeGETRequestJSON(vpnURL + PING_URL_PATH);
+                cameraAddress = json.map(j -> j.getString("local_url"))
+                        .map(localURL -> new CameraAddress(vpnURL, localURL));
+            }
+        }
+        return cameraAddress.map(CameraAddress::getLocalURL);
+    }
+
+    private Optional<JSONObject> executeGETRequestJSON(String url) {
+        Optional<JSONObject> jsonContent = executeGETRequest(url).map(JSONObject::new);
+        if(!jsonContent.isPresent()) {
+            logger.error("The request-result could not get retrieved!");
+        }
+        return jsonContent;
+    }
+
+    Optional<String> executeGETRequest(String url) {
+        try {
+            String content = HttpUtil.executeUrl("GET", url, 5000);
+            if (content != null && !content.isEmpty()) {
+                return Optional.of(content);
+            }
+        } catch (IOException | JSONException e) {
+            logger.error("Error on loading local camera url!", e);
```

Comment:
The error logging level should be reserved for issues catastrophic enough to threaten the operation of openHab itself. An error in a binding shouldn't qualify for this. Please read the [openHab coding guidelines](https://www.openhab.org/docs/developer/guidelines.html#f-logging) for expected use of logging in bindings.

I suggest downgrading this to warning or setting the handler to offline.

Reply:
ok, good hint to the guidelines.

Done.